### PR TITLE
Docs: Introduced new nested site navigation + platform sections

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Update all site navigation to our new tabbed, nested model

Screen Shot 2022-07-28 at 4 37 00 PM
Screen Shot 2022-07-28 at 4 37 09 PM

#### Details

A couple major changes:

1. Rename the docs/components folder to docs/docs-components. This may be a tad redundant, but it avoids confusion with our actual /components/web | ios | android folder
2. Re-do our sidebar index. Becomes `siteIndex` and reorders everything to match the nesting of our docs
3. Switch to using SideNavigation component
4. Add platform switcher to components nav. Uses cookies so it will default to the same platform every time
5. Remove sidenav from Home page
6. Update all URLs on the site to reflect new changes

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-4619)
- [Figma](https://www.figma.com/file/emykswARRS1tQwpB7ThIPN/Gestalt-2022-Site)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

File: docs/components/Header.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 // @flow strict
-import { type Node, useCallback, useEffect, useState, useRef } from 'react';
+import { type Node, useCallback, useEffect, useState, useRef, useMemo } from 'react';
```
Comment (ayeshakmaz) [inline docs/components/Header.js:2]: Main changes here to add Tabs

File: docs/components/Navigation.js [hunk 2]
```diff
@@ -2,74 +2,66 @@
 import { type Node } from 'react';
 import { Box, SideNavigation } from 'gestalt';
 import { useRouter } from 'next/router';
-import sidebarIndex from './sidebarIndex.js';
+import newSidebarIndex, { type siteIndexType } from './siteIndex.js';
```
Comment (ayeshakmaz) [inline docs/components/Navigation.js:5]: Mian changes here to render the side nav with new index and add platform switcher

File: docs/components/useGetSideNavItems.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
+// @flow strict
```
Comment (ayeshakmaz) [inline docs/components/useGetSideNavItems.js:1]: This is the crux of rendering the new sidenav - very open to suggestions here

File: docs/components/siteIndex.js [hunk 4]
```diff
@@ -0,0 +1,166 @@
+// @flow strict
```
Comment (ayeshakmaz) [inline docs/components/siteIndex.js:1]: renamed, and reformatted (still need to update the comment)

File: docs/pages/components/android/overview.js [hunk 5]
```diff
@@ -0,0 +1,14 @@
+// @flow strict
```
Comment (ayeshakmaz) [inline docs/pages/components/android/overview.js:1]: everything here and below is just file structure changes

**Review by AlbertCarreras — APPROVED** [review]

**Pull request merged**